pkg/services/ngalert/store/alert_rule.go

```diff
@@ -270,23 +271,44 @@ func (st DBstore) GetAlertRules(ctx context.Context, query *ngmodels.GetAlertRul
 	})
 }
 
-// GetNamespaces returns the folders that are visible to the user
+// GetNamespaces returns the folders that are visible to the user and have at least one alert in it
```

Comment:
Isn't there a use case for empty folders? How do you put an alert into a folder if you can't get it?

Reply:
This query is not used to get list of folders available for save alerts into. It is used to get a list of folders and use it as filter to get alerts. Please see the code

Reply:
Would a name like `GetUserVisibleNamespaces` reduce this confusion? I could see someone potentially using this in the wrong context, and it guides readers a bit more without needing to parse the method body.

Reply:
> Would a name like GetUserVisibleNamespaces reduce this confusion?

Great idea!